library/ziti.c

```diff
@@ -712,12 +726,20 @@ static void session_cb(ziti_session *session, ziti_error *err, void *ctx) {
         free_ziti_session(old_session);
         FREE(old_session);
 
+        uv_timeval64_t now;
+        uv_gettimeofday(&now);
+
+        int time_diff = (int) (now.tv_sec - session->updated->tv_sec);
+        if (abs(time_diff) > 10) {
+            ZITI_LOG(ERROR, "local clock is %d seconds %s UTC (as reported by controller)", abs(time_diff),
+                     time_diff > 0 ? "ahead" : "behind");
+        }
 
         if (session->expires) {
-            uv_timeval64_t now;
-            uv_gettimeofday(&now);
+            // adjust expiration to local time if needed
+            session->expires->tv_sec += time_diff;
             ZTX_LOG(DEBUG, "ziti API session expires in %ld seconds", (long) (session->expires->tv_sec - now.tv_sec));
-            long delay = (session->expires->tv_sec - now.tv_sec) * 3 / 4;
+            long delay = (session->expires->tv_sec - now.tv_sec) - 10;
```

Comment:
is this on purpose? the - 10?  10 just seems arbitrary ?

Reply:
consistent with golang SDK